sklearn/metrics/tests/test_score_objects.py

```diff
@@ -486,6 +487,8 @@ def check_scorer_memmap(scorer_name):
     scorer, estimator = SCORERS[scorer_name], ESTIMATORS[scorer_name]
     if scorer_name in MULTILABEL_ONLY_SCORERS:
         score = scorer(estimator, X_mm, y_ml_mm)
+    elif scorer_name in NONZERO_Y_SCORERS:
+        score = scorer(estimator, X_mm, y_mm + 1)
```

Comment:
can all use y_mm + 1, actually? Do we even need NONZERO_Y_SCORERS?

Reply:
When trying that, 5 tests cases fail: 

```
======================================================================
ERROR: sklearn.metrics.tests.test_score_objects.test_scorer_memmap_input('balanced_accuracy',)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/usr/lib/python2.7/dist-packages/nose/case.py", line 197, in runTest
    self.test(*self.arg)
  File "/home/dali/Desktop/workspace/scikit-learn/sklearn/utils/testing.py", line 326, in wrapper
    return fn(*args, **kwargs)
  File "/home/dali/Desktop/workspace/scikit-learn/sklearn/metrics/tests/test_score_objects.py", line 493, in check_scorer_memmap
    score = scorer(estimator, X_mm, y_mm + 1)
  File "/home/dali/Desktop/workspace/scikit-learn/sklearn/metrics/scorer.py", line 110, in __call__
    **self._kwargs)
  File "/home/dali/Desktop/workspace/scikit-learn/sklearn/metrics/classification.py", line 1420, in balanced_accuracy_score
    raise ValueError('Balanced accuracy is only meaningful '
ValueError: Balanced accuracy is only meaningful for binary classification problems.

======================================================================
ERROR: sklearn.metrics.tests.test_score_objects.test_scorer_memmap_input('precision',)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/usr/lib/python2.7/dist-packages/nose/case.py", line 197, in runTest
    self.test(*self.arg)
  File "/home/dali/Desktop/workspace/scikit-learn/sklearn/utils/testing.py", line 326, in wrapper
    return fn(*args, **kwargs)
  File "/home/dali/Desktop/workspace/scikit-learn/sklearn/metrics/tests/test_score_objects.py", line 493, in check_scorer_memmap
    score = scorer(estimator, X_mm, y_mm + 1)
  File "/home/dali/Desktop/workspace/scikit-learn/sklearn/metrics/scorer.py", line 110, in __call__
    **self._kwargs)
  File "/home/dali/Desktop/workspace/scikit-learn/sklearn/metrics/classification.py", line 1266, in precision_score
    sample_weight=sample_weight)
  File "/home/dali/Desktop/workspace/scikit-learn/sklearn/metrics/classification.py", line 1044, in precision_recall_fscore_support
    "choose another average setting." % y_type)
ValueError: Target is multiclass but average='binary'. Please choose another average setting.

======================================================================
ERROR: sklearn.metrics.tests.test_score_objects.test_scorer_memmap_input('average_precision',)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/usr/lib/python2.7/dist-packages/nose/case.py", line 197, in runTest
    self.test(*self.arg)
  File "/home/dali/Desktop/workspace/scikit-learn/sklearn/utils/testing.py", line 326, in wrapper
    return fn(*args, **kwargs)
  File "/home/dali/Desktop/workspace/scikit-learn/sklearn/metrics/tests/test_score_objects.py", line 493, in check_scorer_memmap
    score = scorer(estimator, X_mm, y_mm + 1)
  File "/home/dali/Desktop/workspace/scikit-learn/sklearn/metrics/scorer.py", line 211, in __call__
    return self._sign * self._score_func(y, y_pred, **self._kwargs)
  File "/home/dali/Desktop/workspace/scikit-learn/sklearn/metrics/ranking.py", line 217, in average_precision_score
    sample_weight=sample_weight)
  File "/home/dali/Desktop/workspace/scikit-learn/sklearn/metrics/base.py", line 75, in _average_binary_score
    return binary_metric(y_true, y_score, sample_weight=sample_weight)
  File "/home/dali/Desktop/workspace/scikit-learn/sklearn/metrics/ranking.py", line 209, in _binary_uninterpolated_average_precision
    y_true, y_score, sample_weight=sample_weight)
  File "/home/dali/Desktop/workspace/scikit-learn/sklearn/metrics/ranking.py", line 470, in precision_recall_curve
    sample_weight=sample_weight)
  File "/home/dali/Desktop/workspace/scikit-learn/sklearn/metrics/ranking.py", line 364, in _binary_clf_curve
    raise ValueError("Data is not binary and pos_label is not specified")
ValueError: Data is not binary and pos_label is not specified

======================================================================
ERROR: sklearn.metrics.tests.test_score_objects.test_scorer_memmap_input('recall',)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/usr/lib/python2.7/dist-packages/nose/case.py", line 197, in runTest
    self.test(*self.arg)
  File "/home/dali/Desktop/workspace/scikit-learn/sklearn/utils/testing.py", line 326, in wrapper
    return fn(*args, **kwargs)
  File "/home/dali/Desktop/workspace/scikit-learn/sklearn/metrics/tests/test_score_objects.py", line 493, in check_scorer_memmap
    score = scorer(estimator, X_mm, y_mm + 1)
  File "/home/dali/Desktop/workspace/scikit-learn/sklearn/metrics/scorer.py", line 110, in __call__
    **self._kwargs)
  File "/home/dali/Desktop/workspace/scikit-learn/sklearn/metrics/classification.py", line 1364, in recall_score
    sample_weight=sample_weight)
  File "/home/dali/Desktop/workspace/scikit-learn/sklearn/metrics/classification.py", line 1044, in precision_recall_fscore_support
    "choose another average setting." % y_type)
ValueError: Target is multiclass but average='binary'. Please choose another average setting.

======================================================================
ERROR: sklearn.metrics.tests.test_score_objects.test_scorer_memmap_input('f1',)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/usr/lib/python2.7/dist-packages/nose/case.py", line 197, in runTest
    self.test(*self.arg)
  File "/home/dali/Desktop/workspace/scikit-learn/sklearn/utils/testing.py", line 326, in wrapper
    return fn(*args, **kwargs)
  File "/home/dali/Desktop/workspace/scikit-learn/sklearn/metrics/tests/test_score_objects.py", line 493, in check_scorer_memmap
    score = scorer(estimator, X_mm, y_mm + 1)
  File "/home/dali/Desktop/workspace/scikit-learn/sklearn/metrics/scorer.py", line 110, in __call__
    **self._kwargs)
  File "/home/dali/Desktop/workspace/scikit-learn/sklearn/metrics/classification.py", line 717, in f1_score
    sample_weight=sample_weight)
  File "/home/dali/Desktop/workspace/scikit-learn/sklearn/metrics/classification.py", line 831, in fbeta_score
    sample_weight=sample_weight)
  File "/home/dali/Desktop/workspace/scikit-learn/sklearn/metrics/classification.py", line 1044, in precision_recall_fscore_support
    "choose another average setting." % y_type)
ValueError: Target is multiclass but average='binary'. Please choose another average setting.

----------------------------------------------------------------------
Ran 53 tests in 0.244s

FAILED (errors=5)
```

Reply:
Also the type of `y_mm` is `<class 'numpy.core.memmap.memmap'>` whereas `y_mm + 1` is of type 
`<type 'numpy.ndarray'>`

Reply:
Indeed. Well it's not a useful test if it's not a memmap. Make sure to store it with +1 ...